src/java.desktop/macosx/classes/com/apple/laf/AquaKeyBindings.java

```diff
@@ -90,6 +90,12 @@ void setDefaultAction(final String keymapName) {
         "alt KP_LEFT", DefaultEditorKit.previousWordAction,
         "alt RIGHT", DefaultEditorKit.nextWordAction,
         "alt KP_RIGHT", DefaultEditorKit.nextWordAction,
+        "alt altgraph LEFT", DefaultEditorKit.previousWordAction,
+        "alt altgraph RIGHT", DefaultEditorKit.nextWordAction,
+        "alt UP", DefaultEditorKit.beginLineUpAction,
```

Comment:
I find it utterly bizarre to say we need "altgraph" when macOS doesn't have altgraph - certainly not on the k/bs on current mac books or imacs.

My take is that the previous fix JDK-8151136 was a mistake and we should have just excluded the test there on macOS.

Reply:
I also agree.

Reply:
That's not a mistake, the swing was updated in a few places here and there to support it, for example:
https://bugs.openjdk.org/browse/JDK-8194873

Reply:
We could change it back but then we should take care of all that followup changes.

Reply:
> That's not a mistake, the swing was updated in a few places here and there to support it, for example: https://bugs.openjdk.org/browse/JDK-8194873

I dont think @prrace and me are contesting need of ALT_GRAPH in swing, which was fixed by the above bug..I guess it was being contested for macos only so we are not going to revert 8194873 but only 8151136 (which was done to convert right alt to altgraph for macos)

Reply:
That is similar issue on Windows since we unconditionally set the right alt key to be ALT_GRAPH independently on the current keyboard layout, it is just a button location which is the same on win and mac. So probably right now it is better to update the spec of that flag. 

To workaround that change the fix for [JDK-8194873](https://bugs.openjdk.org/browse/JDK-8194873) updated all our L&F which are used on windows, obviously the Aqua was missed. I think we should update the Aqua in a similar way.

Reply:
I can see "AltGr" key in windows keyboard but not mac so am not sure why should we support just because of key location.
Anyway, I will let @prrace decide  on that.

If we need to support altgr, then present PR supports both 8151136  (alt graph) and also Right/Left Option Up/Down

Reply:
One macOS you will not see the "Alt" key as well, but we use that name.

Reply:
> One macOS you will not see the "Alt" key as well, but we use that name.

At least I can see
![mac-keyboard-alt](https://user-images.githubusercontent.com/43534309/177146818-c4c17e1d-5f61-4c9c-9122-947606fc59c1.png)

Reply:
As per offline discussion with @prrace ALT_GRAPH support done via JDK-8151136 is reverted and tests testing ALT_GRAPH in macos is excluded..
@prrace can you please review?

Reply:
Even if JavaFX, Left and Right Option key is mapped to ALT so I guess we will be consistent with that too.

Reply:
> > One macOS you will not see the "Alt" key as well, but we use that name.
> 
> At least I can see 

Just a clarification:

On the new keyboards, it does not have it, I have two laptops where the "alt" label is missing:
![mac-keyboard-alt](https://support.apple.com/library/content/dam/edam/applecare/images/en_US/social/keyboards-social-card.jpg)

The official document about the keyboard [mentions ](https://support.apple.com/en-us/HT201236) the "alt" key in case the windows keyboard is used: `"On keyboards made for Windows PCs, use the Alt key instead of Option, and the Windows logo key instead of Command."`

So I do not see a reason why the "right option" - "alt gr" key on the PC keyboard cannot be mapped to the ALT_GRAPH, same as "left option" - "alt" is mapped to the ALT.

Reply:
The document mentioned "use the Alt key", which is what both Option keys are mapped to now. I dont see anywhere it is mentioned to use "AltGraph" key, so I dont see the need of mapping Right Option to Alt Graph....I dont see any native app where Right Option is being used as "AltGraph"..
As I mentioned, even FX maps both keys to Alt...